### PR TITLE
[SYCL] Add support for regular pointers to math functions for USM.

### PR DESCRIPTION
**Pull request by jcranmer-intel**
Signed-off-by: Joshua Cranmer <joshua.cranmer@intel.com>

### CODE REVIEW

File: sycl/test/usm/math.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
```
Comment (alexeyvoronov) [inline sycl/test/usm/math.cpp]: Does this functionality work with host device? If yes, please add run on host.
`// RUN: env SYCL_DEVICE_TYPE=HOST %t.out`

File: sycl/test/usm/math.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  {
+    s::cl_float r{ 0 };
+    s::cl_float i{ 999 };
+    {
+      s::cl_float *Buf = (s::cl_float *)s::malloc_shared(sizeof(cl_float) * 2,
+          myQueue.get_device(), myQueue.get_context());
+      s::malloc_shared(100, myQueue.get_device(), myQueue.get_context());
+      myQueue.submit([&](s::handler &cgh) {
+        cgh.single_task<class fractF1UF1>([=]() {
+          Buf[0] = s::fract(s::cl_float{ 1.5f }, &Buf[1]);
```
Comment (alexeyvoronov) [inline sycl/test/usm/math.cpp]: Can you also test these built-in' functions with vector type?

File: sycl/test/usm/math.cpp (hunk 1 above)
Comment (jcranmer-intel) [reply sycl/test/usm/math.cpp]: I copied the test environments from other USM tests, which only test the CPU_RUN_PLACEHOLDER environment. I can try the host device and add it if it works.

Comment (alexeyvoronov) [reply sycl/test/usm/math.cpp]: I did not work with USM, but I believe that it should not break the work with the host device, and therefore should be tested.

Comment (jcranmer-intel) [reply sycl/test/usm/math.cpp]: I tested it with the host device, and it did fail. James says that the host device doesn't support it at the moment.

File: sycl/test/usm/math.cpp (hunk 2 above)
Comment (jcranmer-intel) [reply sycl/test/usm/math.cpp]: Added. The code didn't work at first, so I had to do some more modification to get it work. Now I really wish I could use constexpr if...

**Pull request merged**